### PR TITLE
Only sort keys on objects

### LINKED ISSUE #20: let's add input type checks for "data" argument
Thank you for creating and maintaining this library. I want to report a following issue.

Currently, "data" argument is checked only for existence:
https://github.com/sindresorhus/write-json-file/blob/master/index.js#L15

If it exists, it is assumed that is it _a plain object_. The problem is, accidentally the `data` can come as `string` too and this library will silently write **an empty file**! For example, [format-package](https://www.npmjs.com/package/format-package) returns a string and that's why `write-json-file` in one of my libraries was silently writing empty files.

Thinking about this, we could either:

1. always _throw_ if `data` is not a plain object, 

or, more cheeky option, do this: 

2. if there are any processing options set, attempt to parse the string, try-catch and throw a meaningful error message if it's not a JSON string; ELSE (no processing), pipe all inputs straight into `write-file-atomic`, skipping `JSON.stringify`.

Arguments for the second option are the following.

* Maybe somebody already has stringified JSON and just wants to write it do disk? Installing `write-file-atomic` as a new dependency is _tedious_, plus, it's a possible _liability_ if directly tapped version of `write-file-atomic` mismatches the one that `write-json-file` uses. Promisifying `write-file-atomic` is another liability — what if user does not read the source and tap another promisifying library, other than `pify`?

* In `format-package` case where data is string, it would be terribly inefficient to `JSON.parse` the data only to be right away stringified by `write-json-file`.

* Second option is also more user-friendly: we could attempt to parse `data` (if processing is requested) and that would not require any extra dependencies.. And if there is no processing requested, we could cut corner and skip `JSON.stringify`.

* Also, the first option would require plain object checks which is a new dependency like `lodash.isplainobject`. Going 2nd option route we avoid this, we just attempt to parse if string is given, else `throw`. That's no extra dependencies.

Either way, even if we do nothing about this, at least let's **write a unit test** and cover the case, what happens when `data` is: **1) empty string, 2) non-empty string but not JSON, 3) stringified valid JSON**.

### PR DESCRIPTION
**Pull request by kevva**
Fixes #20.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,7 +22,7 @@ const init = (fn, filePath, data, options) => {
 		sortKeys: false
 	}, options);
 
-	if (options.sortKeys) {
+	if (options.sortKeys && isPlainObj(data)) {
```
Comment (sindresorhus) [inline index.js]: People are allowed to use any kind of object with `JSON.stringify()`, not just plain objects, so I don't think we should be this strict.

Comment (sindresorhus) [reply index.js]: Better to use https://github.com/sindresorhus/is-obj

Comment (kevva) [reply index.js]: `sort-keys` itself uses `is-plain-obj` though so not sure how to go about it?

Comment (sindresorhus) [reply index.js]: Good point 👍

**Pull request merged**